lib/dependencies/AMDDefineDependency.js

```diff
@@ -2,113 +2,137 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var NullDependency = require("./NullDependency");
+"use strict";
+const NullDependency = require("./NullDependency");
 
-function AMDDefineDependency(range, arrayRange, functionRange, objectRange) {
-	NullDependency.call(this);
-	this.range = range;
-	this.arrayRange = arrayRange;
-	this.functionRange = functionRange;
-	this.objectRange = objectRange;
+class AMDDefineDependency extends NullDependency {
+	constructor(range, arrayRange, functionRange, objectRange) {
+		super();
+		this.range = range;
+		this.arrayRange = arrayRange;
+		this.functionRange = functionRange;
+		this.objectRange = objectRange;
+	}
+
+	get type() {
+		return "amd define";
+	}
 }
-module.exports = AMDDefineDependency;
 
-AMDDefineDependency.prototype = Object.create(NullDependency.prototype);
-AMDDefineDependency.prototype.constructor = AMDDefineDependency;
-AMDDefineDependency.prototype.type = "amd define";
+AMDDefineDependency.Template = class AMDDefineDependencyTemplate {
+	get definitions() {
+		return {
```

Comment:
I meant
```js
AMDDefineDependency.prototype.type = "amd define";
```
Will be more performant because V8's Crankshaft will optimise this. If you'll have `const definitions` it won't create a hidden class because of closure (actually maybe it will in latest versions if it's smart enough to find out that `definitions` is a string literal). Anyway I'm not opposed to using closure since I don't have any benchmarks (but we should create them to improve Webpack performance after transition to ES6+ will be done).